lightning/src/routing/router.rs

```diff
@@ -2225,7 +2313,32 @@ where L::Target: Logger {
 	for results_vec in selected_paths {
 		let mut hops = Vec::with_capacity(results_vec.len());
 		for res in results_vec { hops.push(res?); }
-		paths.push(Path { hops, blinded_tail: None });
+
+		let blinded_path = payment_params.payee.blinded_route_hints().iter().enumerate()
+			.find(|(idx, (_, p))| {
+				DUMMY_BLINDED_SCID + *idx as u64 == hops[hops.len() - 1].short_channel_id ||
+					(p.blinded_hops.len() == 1 && p.introduction_node_id == hops[hops.len() - 1].pubkey)
+			})
+			.map(|(_, (_, p))| p.clone());
+		let blinded_tail = if let Some(BlindedPath { blinded_hops, blinding_point, .. }) = blinded_path {
+			let num_blinded_hops = blinded_hops.len();
+			Some(BlindedTail {
+				hops: blinded_hops,
+				blinding_point,
+				excess_final_cltv_expiry_delta: 0,
+				final_value_msat: {
+					if num_blinded_hops > 1 {
+						hops.pop().unwrap().fee_msat
```

Comment:
Just want to make sure the `pop` is intentional here.

Reply:
Yeah, added a comment